### PR TITLE
feat(#117): Add support for customization of user fields in the api

### LINKED ISSUE #117: Better UserSerializer
### Environment
* Python version: 3.8
* django-comments-dab version: 2.5.1 

### Proposed Functionality
The get_user_fields function used in UserSerializer has the hardcoded field of username. In some projects we don't use username as the primary field. I had to change the function to make it work with my software.

https://github.com/Radi85/Comment/blob/1c2359e2c11f6efa4d62ba4d260848b3c8e37ead/comment/api/serializers.py#L22-L28

I suggest something like this. 

```python
def get_user_fields():
    user_model = get_user_model()
    fields = user_model._meta.get_fields()
    for field in fields:
        if hasattr(field, "upload_to"):
            return list(set('id', user_model.USERNAME_FIELD, 'email', 'profile', field.name))
    return 'id', 'email', 'profile'
```

### Use Case
API will work with projects that have email based or phone based authentication

### Database Changes
No Database changes required

### PR DESCRIPTION
**Pull request by abhiabhi94**
- resolves #117
- closes #126

### CODE REVIEW

File: comment/api/serializers.py [hunk 1]
```diff
@@ -23,10 +26,13 @@ def get_profile_model():
 def get_user_fields():
     user_model = get_user_model()
     fields = user_model._meta.get_fields()
+    api_fields = list(settings.COMMENT_USER_API_FIELDS) + ['profile']
+    # remove duplicates, preserve order
+    api_fields = list(OrderedDict.fromkeys(api_fields))
```
Comment (Radi85) [inline comment/api/serializers.py:31]: Please use `set` for removing duplication.
- More readability.
- No extra import statement needed.
- No comment needed to explain what this line is doing.

```suggestion
    api_fields = list(set(api_fields))
```

**Review by Radi85 — CHANGES_REQUESTED** [review]

File: comment/api/serializers.py (hunk 1 above)
Comment (abhiabhi94) [reply comment/api/serializers.py:31]: `sets` don't necessarily prevent orders and that is what the comment says. we would want to preserve the order in which custom fields have been specified.

Comment (Radi85) [reply comment/api/serializers.py:31]: I am not sure that I understood what you mean, but why would we care about the orders? We are getting unique values from a list, right?

Comment (abhiabhi94) [reply comment/api/serializers.py:31]: i have set the fields to `[`a`, 'b']` and now my frontend code is written in a way that tends to care about the order in which these fields are returned.

Comment (Radi85) [reply comment/api/serializers.py:31]: Hmmm! You might be confused here? The returned value is json not an ordered list, so these fields will be used in json object like so:
```python
fields = ['a', 'b']

response = {
    'a': a_value,
    'b': b_value
}
```

That means keeping the same order has no meaning.

**Review by Radi85 — CHANGES_REQUESTED** [review]
Docs need to be updated accordingly!

**Pull request merged**